src/components/Topic.jsx

```diff
@@ -0,0 +1,57 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+
+import axios from "axios";
+
+function Topic() {
+  const { single_topic } = useParams();
+  const [topicList, setTopicList] = useState([]);
```

Comment:
I think the naming of you variables here is a bit confusing

`topicList` sounds like a list of topics (`['cooking', 'football', 'coding']`) but it's actually a list of articles!